src/resolvers/user.ts

```diff
@@ -1,36 +1,322 @@
     users: async (_, __, context): Promise<User[]> => {
-      // TODO: remove this hard-coded affiliationId once the User is in the context
-      const affiliationId = 'https://ror.org/01nrxwf90';
-      return await User.findByAffiliationId('users resolver', context, affiliationId)
+      if (isAdmin(context.token)) {
+        return await User.findByAffiliationId('users resolver', context, context.token.affiliationId);
+      }
+      // Unauthorized!
+      throw context?.token ? ForbiddenError() : AuthenticationError();
     },
 
     // This query should only be available to Admins. Super can get any user and Admin can get
     // only users associated with their affiliationId
     user: async (_, { userId }, context: MyContext): Promise<User> => {
-      return await User.findById('user resolver', context, userId);
+      if (isAdmin(context.token)) {
+        const user = await User.findById('user resolver', context, userId);
+        if (!user) {
+          throw NotFoundError();
+        }
+        // Make sure the Admin is from the same Affiliation or the user is a SuperAdmin
+        if (context.token?.affiliationId === user.affiliationId || await isSuperAdmin(context.token)) {
+          return user;
+        }
+      }
+      // Unauthorized!
+      throw context?.token ? ForbiddenError() : AuthenticationError();
+    },
+  },
+
+  Mutation: {
+    // Update the current user's information
+    updateUserProfile: async (_, { input: {
+      givenName,
+      surName,
+      affiliationId,
+      languageId,
+    } }, context: MyContext): Promise<User> => {
+      if (isAuthorized(context?.token)) {
+        const user = await User.findById('updateUserProfile resolver', context, context.token.id);
+        // Only continue if the user is active and not locked
+        if (!user || !user.active || user.locked) {
+          throw ForbiddenError();
+        }
+
+        user.givenName = givenName;
+        user.surName = surName;
+        user.affiliationId = affiliationId;
+        user.languageId = languageId || defaultLanguageId;
+        const updated = await new User(user).update(context);
+        if (!updated) {
+          throw InternalServerError('Unable to save the profile changes at this time');
+        }
+        return updated;
+      }
+      // Unauthenticated
+      throw AuthenticationError();
+    },
+    // Update the current user's email notifications
+    updateUserNotifications: async (_, { input: {
+      notify_on_comment_added,
+      notify_on_template_shared,
+      notify_on_feedback_complete,
+      notify_on_plan_shared,
+      notify_on_plan_visibility_change,
+    } }, context: MyContext): Promise<User> => {
+      if (isAuthorized(context?.token)) {
+        const user = await User.findById('updateUserNotifiactions resolver', context, context.token.id);
+        // Only continue if the user is active and not locked
+        if (!user || !user.active || user.locked) {
```

Comment:
L88: Not a big deal, but I noticed this is used multiple times, and was wondering if it'd make sense to put it in a separate function?

Reply:
Yes. I am going to create a ticket to add some helper methods for some duplicative code. I was thinking about it recently. Like our `if (updated && (updated.errors && Array.isArray(updated.errors) ...`

Reply:
Thanks Brian.